ui/src/singletons/store.ts

```diff
@@ -245,7 +245,7 @@ const useStore = create<AppStore>()(
         },
       }),
       {
-        limit: 50,
+        limit: 20,
```

Comment:
I'd say let's stay generous and keep this limit at 50 until we can prove it causes an issue.